apis/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -166,5 +167,12 @@ func (r *OpenTelemetryCollector) validateCRDSpec() error {
 
 	}
 
+	mode := strings.ToLower(string(r.Spec.Mode))
+	if r.Spec.Ingress.Type == IngressTypeNginx && (mode == "deployment" || mode == "daemonset" || mode == "statefulset") {
+		return fmt.Errorf("the OptenTelemetry Spec Ingress configuiration is incorrect. Ingress can only be used in combination with the modes: %s, %s, %s",
+			"deployment", "daemenset", "statefulset",
```

Comment:
these strings can be directly inlined into the message. did you intend to use the mode constants?